sessionctx/variable/sysvar.go

```diff
@@ -864,7 +864,7 @@ var defaultSysVars = []*SysVar{
 		s.HashExchangeWithNewCollation = TiDBOptOn(val)
 		return nil
 	}},
-	{Scope: ScopeSession, Name: TiDBEnforceMPPExecution, Type: TypeBool, Value: BoolToOnOff(config.GetGlobalConfig().Performance.EnforceMPP), Validation: func(vars *SessionVars, normalizedValue string, originalValue string, scope ScopeFlag) (string, error) {
+	{Scope: ScopeGlobal | ScopeSession, Name: TiDBEnforceMPPExecution, Type: TypeBool, Value: BoolToOnOff(config.GetGlobalConfig().Performance.EnforceMPP), Validation: func(vars *SessionVars, normalizedValue string, originalValue string, scope ScopeFlag) (string, error) {
 		if TiDBOptOn(normalizedValue) && !vars.allowMPPExecution {
 			return normalizedValue, ErrWrongValueForVar.GenWithStackByArgs("tidb_enforce_mpp", "1' but tidb_allow_mpp is 0, please activate tidb_allow_mpp at first.")
 		}
```

Comment:
This behavior is now weird, since `vars.allowMPPExecution` is the session scope value. If you are validating for global scope, you should check the matching global value instead.

Here is an example where this is done this way: https://github.com/pingcap/tidb/blob/4c49e5204fb60b01a76b78619eb6627a1a387a71/sessionctx/variable/varsutil.go#L132-L150